app/services/query.js

```diff
@@ -215,9 +215,7 @@ export default Ember.Service.extend({
      */
   eventCandidates(params) {
     const candidates = this.get('ajax').request('/datasets', {data: params});
-    return candidates.then(doc => {
-      return doc.objects.map(this.camelizeHash);
-    }, function(reason) {
+    return candidates.then(doc => doc.objects.map(v => this.injectExplorerURL("event", params, this.camelizeHash(v))), function(reason) {
```

Comment:
I like the solution here, but two anonymous functions plus launching an error-handling function is a lot to take in one line. Take pity on my week mind when formatting your code.